public/src/overmind/state.tsx

```diff
@@ -297,46 +310,46 @@ export const state: State = {
                         }
                         return sortOrder
                     }
-                    const aSubs = a.submissions ? getSubmissionsScore(a.submissions) : 0
-                    const bSubs = b.submissions ? getSubmissionsScore(b.submissions) : 0
+                    const aSubs = subsA ? getSubmissionsScore(subsA) : 0
+                    const bSubs = subsB ? getSubmissionsScore(subsB) : 0
                     return sortOrder * (aSubs - bSubs)
                 }
                 case SubmissionSort.Approved: {
-                    if (rootState.review.assignmentID > 0) {
+                    if (assignmentID > 0) {
                         const sA = subA && isApproved(subA) ? 1 : 0
                         const sB = subB && isApproved(subB) ? 1 : 0
                         return sortOrder * (sA - sB)
                     }
-                    const aApproved = a.submissions ? getNumApproved(a.submissions) : 0
-                    const bApproved = b.submissions ? getNumApproved(b.submissions) : 0
+                    const aApproved = subsA ? getNumApproved(subsA) : 0
+                    const bApproved = subsB ? getNumApproved(subsB) : 0
                     return sortOrder * (aApproved - bApproved)
                 }
                 case SubmissionSort.Name: {
-                    const nameA = a.user?.Name ?? ""
-                    const nameB = b.user?.Name ?? ""
+                    const nameA = groupView ? a.name ?? "" : a.user?.Name ?? ""
+                    const nameB = groupView ? b.name ?? "" : b.user?.Name ?? ""
                     return sortOrder * (nameA.localeCompare(nameB))
                 }
                 default:
                     return 0
             }
         })
-        return sortedSubmissions
-    }),
-    activeSubmission: derived((state: State) => {
-        if (state.activeSubmissionLink) {
-            return state.activeSubmissionLink.submission ? Number(state.activeSubmissionLink.submission.ID) : -1
-        }
-        return -1
+        return sortedSubmissions as Group[] | Enrollment[]
     }),
+    activeSubmission: 0n,
     activeEnrollment: null,
-    activeSubmissionLink: null,
-    currentSubmission: derived(({ activeSubmissionLink }: State) => {
-        return activeSubmissionLink?.submission ?? null
+    currentSubmission: derived(({ activeSubmission, submissionsForCourse, submissionOwner }: State) => {
```

Comment:
It may be a bit inefficient, but couldn't we replace `submissionOwner` and the `ForOwner` method here with a new method on `submissionsForCourse` that examines all submissions in both user and group maps for a specific submission ID. For example, `submissionsForCourse.Submission(activeSubmission)`?